massa-ledger-worker/src/ledger_db.rs

```diff
@@ -330,13 +330,13 @@ impl LedgerDB {
     ///
     /// # Returns
     /// A `BTreeSet` of the datastore keys
-    pub fn get_datastore_keys(&self, addr: &Address) -> BTreeSet<Vec<u8>> {
+    pub fn get_datastore_keys(&self, addr: &Address) -> Option<BTreeSet<Vec<u8>>> {
         let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
 
         let mut opt = ReadOptions::default();
         opt.set_iterate_upper_bound(end_prefix(data_prefix!(addr)).unwrap());
 
-        self.db
+        Some(self.db
```

Comment:
I think this will still return Some even if the address does not exist in the ledger (?)

Reply:
We might need your help here @Eitu33

Reply:
> I think this will still return Some even if the address does not exist in the ledger (?)

Yes it will return `Some([])` in that case

Reply:
If the required behaviour is `None` when the addr isn't in the ledger you should check for its balance or bytecode entry before, can't use the datastore size as indicator as it will have the same result for non-existant addr and addr with no datatstore

Reply:
Ok then let's look for the balance

Reply:
Yeah we could set the rockdb iterator lower bound to the address balance prefix identifier which would include the balance and bytecode in the retrieved keys, and then handle it as you wish

Reply:
Nice ! Can you sync on that together ?

Reply:
Seen with Thomas so I've impl a 'quick fix' so we can merge this for TESTNET 17 and I'd like to create an issue to impl a better version (with rockdb iterator). Is this ok for you @damip ?

Reply:
sounds good !